edb/edgeql/parser/grammar/ddl.py

```diff
@@ -1638,6 +1641,33 @@ def reduce_DropFunction(self, *kids):
             params=kids[3].val)
 
 
+#
+# ALTER FUNCTION
+#
+
+commands_block(
+    'AlterFunction',
+    SetFieldStmt,
+    CreateAnnotationValueStmt,
+    AlterAnnotationValueStmt,
+    DropAnnotationValueStmt,
+    opt=False
+)
+
+
+class AlterFunctionStmt(Nonterm):
```

Comment:
You also need to reflect changes to volatility in Postgres by adding code to `pgsql.delta.AlterFunction`.  The check would be `if self.get_attribute_value('volatility'): <code-as-in-CreateFunction>` (with `dbops.CreateFunction(func, or_replace=True)`).  Incidentally, you should be able to support body changes this way as well.

Reply:
Good point. I'll see if I can add some testmode stuff to introspect Postgres volatility and test this thoroughly.